src/blockchain/database/common/BlockHeaders.cpp

```diff
@@ -109,44 +103,38 @@ auto BlockHeader::store(
     const auto& hash = header.Hash();
 
     try {
-        const auto proto = [&] {
-            auto out = block::internal::Header::SerializedType{};
+        block::internal::Header::SerializedType proto{};
 
-            if (false == header.Internal().Serialize(out)) {
-                throw std::runtime_error{"Failed to serialized header"};
-            }
+        if (false == header.Internal().Serialize(proto)) {
+            throw std::runtime_error{"Failed to serialized header"};
+        }
 
-            if (clearLocal) { out.clear_local(); }
+        if (clearLocal) { proto.clear_local(); }
 
-            return out;
-        }();
         const auto bytes = proto.ByteSizeLong();
-        auto index = [&] {
-            auto output = util::IndexData{};
-            auto cb = [&output](const ReadView in) {
-                if (sizeof(output) != in.size()) { return; }
-
-                std::memcpy(static_cast<void*>(&output), in.data(), in.size());
-            };
-            lmdb_.Load(table_, hash.Bytes(), cb);
-
-            return output;
-        }();
-        auto cb = [&](auto& tx) -> bool {
+
+        util::IndexData index{};
+        auto cb = [&index](const ReadView in) {
+            if (sizeof(index) != in.size()) { return; }
+
+            std::memcpy(static_cast<void*>(&index), in.data(), in.size());
+        };
+        lmdb_.Load(table_, hash.Bytes(), cb);
+
+        auto callback = [&](auto& tx) -> bool {
```

Comment:
The name change (and a couple of name changes before) could be avoided with a pair of braces to limit the scope...

Reply:
I will keep it like this, since Im gonna move lmdb_..Load stuff to separate function. It is used quite often in common so no need to duplicate code